connector.go

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2020 Snowflake Computing Inc. All right reserved.
+
+package gosnowflake
+
+import (
+	"context"
+	"database/sql/driver"
+)
+
+// InternalSnowflakeDriver is the interface for an internal Snowflake driver
+type InternalSnowflakeDriver interface {
```

Comment:
I tried to move this interface definition into a private package `gosnowflake/private` but because this need to refer back to `Config`, there was a cyclical dependency unless we also move `Config`, but that would break existing back compat.

So I prefixed this as `InternalSnowflakeDriver`